lizmap/modules/lizmap/lib/Project/QgisProject.php

```diff
@@ -378,33 +378,54 @@ protected function setShortNames(ProjectConfig $cfg)
     protected function setLayerOpacity(ProjectConfig $cfg)
     {
         $layers = $this->layers;
-        // loop through project layers to get embedded ones as well
-        foreach ($layers as $layer) {
-            $xpath = "//maplayer[id='{$layer['id']}']/layerOpacity[.!=1]/parent::* | //maplayer[id='{$layer['id']}']/pipe/rasterrenderer/@opacity[.!=1]/ancestor::maplayer";
-            $layerWithOpacity = null;
-
-            // check if layer is embedded or not
-            $qgisProject = $this->getEmbeddedQgisProject($layer['id']);
-
-            if ($qgisProject) {
-                $layerWithOpacity = $qgisProject->xpathQuery($xpath);
-            } else {
-                $layerWithOpacity = $this->xpathQuery($xpath);
-            }
-            if ($layerWithOpacity && count($layerWithOpacity) == 1) {
-                $name = (string) $layerWithOpacity[0]->layername;
+        $xpathOpacity = '//maplayer/layerOpacity[.!=1]/parent::* | //maplayer/pipe/rasterrenderer/@opacity[.!=1]/ancestor::maplayer';
+        // non-embedded layers
+        $layerWithOpacities = $this->xpathQuery($xpathOpacity);
+        if ($layerWithOpacities) {
+            foreach ($layerWithOpacities as $layerWithOpacity) {
+                $name = (string) $layerWithOpacity->layername;
                 $layerCfg = $cfg->getLayer($name);
                 $opacity = 1;
                 if ($layerCfg) {
-                    if (isset($layerWithOpacity[0]->layerOpacity)) {
-                        $opacity = (float) $layerWithOpacity[0]->layerOpacity;
-                    } elseif (isset($layerWithOpacity[0]->pipe->rasterrenderer['opacity'])) {
-                        $opacity = (float) $layerWithOpacity[0]->pipe->rasterrenderer['opacity'];
+                    if (isset($layerWithOpacity->layerOpacity)) {
+                        $opacity = (float) $layerWithOpacity->layerOpacity;
+                    } elseif (isset($layerWithOpacity->pipe->rasterrenderer['opacity'])) {
+                        $opacity = (float) $layerWithOpacity->pipe->rasterrenderer['opacity'];
                     }
                     $layerCfg->opacity = $opacity;
                 }
             }
         }
+
+        // embedded layers contains id
+        foreach ($layers as $layer) {
+            $xpathEmbedded = "//maplayer[@id='{$layer['id']}']";
+            $embeddedLayer = $this->xpathQuery($xpathEmbedded);
+            if ($embeddedLayer) {
```

Comment:
Maybe it's only me, but I prefer when we negate checks like this.

```
if not something:
     continue
```

so we can keep code on the left side of the editor and not moving the logic to the right. And we are sure there isn't code executer under the the `if`.

My 2 cents, feel free to skip.

Reply:
agree